### PR TITLE
Channel executor not injected

### LINKED ISSUE #5541: Non-documented breaking change of channel-executor configuration
**Version Information**
Version of Akka.NET? 1.4.32
Which Akka.NET Modules? Akka

**Describe the bug**
After update from 1.4.20 to 1.4.32 channel-executor configuration (see [doc](https://getakka.net/articles/actors/dispatchers.html#channelexecutor)) is not valid anymore

**To Reproduce**
1. Create a console app with Akka 1.4.29+
2. Execute:
```
using Akka.Actor;

namespace ChannelExecutorIssue
{
    internal class Program
    {
        static void Main(string[] args)
        {
            // Following: https://getakka.net/articles/actors/dispatchers.html#channelexecutor
            var config = @"
                          akka.actor.default-dispatcher = { 
                                 executor = channel-executor
                          }";
            
            // Throws NRE in 1.4.29+
            var actorSystem = ActorSystem.Create("test", config);
        }
    }
}
```
3. Observe

**Expected behavior**
ActorSystem created successfully

**Actual behavior**
NRE is thrown:
```
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at Akka.Dispatch.ChannelExecutorConfigurator..ctor(Config config, IDispatcherPrerequisites prerequisites)
   at Akka.Dispatch.MessageDispatcherConfigurator.ConfigureExecutor()
   at Akka.Dispatch.DispatcherConfigurator..ctor(Config config, IDispatcherPrerequisites prerequisites)
   at Akka.Dispatch.Dispatchers.ConfiguratorFrom(Config cfg)
   at Akka.Dispatch.Dispatchers.LookupConfigurator(String id)
   at Akka.Dispatch.Dispatchers.Lookup(String dispatcherName)
   at Akka.Dispatch.Dispatchers..ctor(ActorSystem system, IDispatcherPrerequisites prerequisites, ILoggingAdapter logger)
   at Akka.Actor.Internal.ActorSystemImpl.ConfigureDispatchers()
   at Akka.Actor.Internal.ActorSystemImpl..ctor(String name, Config config, ActorSystemSetup setup, Nullable`1 guardianProps)
   at Akka.Actor.ActorSystem.CreateAndStartSystem(String name, Config withFallback, ActorSystemSetup setup)
   at Akka.Actor.ActorSystem.Create(String name, Config config)
   at ChannelExecutorIssue.Program.Main(String[] args)
```
   

The issue is in the default config for `default-dispatcher` that has no `channel-executor.priority` set. Introduced in #5403    

Sidenote: default `channel-scheduler` configuration is expected to be at: `akka.channel-scheduler` https://github.com/akkadotnet/akka.net/blob/ccb467076f061239aecc976983c6990c372811dd/src/core/Akka/Dispatch/ChannelSchedulerExtension.cs#L63
But it is actually located under `akka.actor.channel-scheduler`     
Current workaround:
```
using Akka.Actor;

namespace ChannelExecutorIssue
{
    internal class Program
    {
        static void Main(string[] args)
        {
            //Copy akka.actor.channel-scheduler from Akka/Configuration/Pigeon.conf
            var config = @"
                          akka.channel-scheduler {
                              parallelism-min = 4    #same as for ForkJoinDispatcher
                              parallelism-factor = 1 #same as for ForkJoinDispatcher
                              parallelism-max = 64   #same as for ForkJoinDispatcher
                              work-max = 10          #max executed work items in sequence until priority loop
                              work-interval = 500    #time target of executed work items in ms
                              work-step = 2          #target work item count in interval / burst
                          }
                          akka.actor.default-dispatcher = { 
                              executor = channel-executor
                              channel-executor.priority = normal
                          }";
            
            var actorSystem = ActorSystem.Create("test", config);
        }
    }
}
```

### PR DESCRIPTION
**Pull request by eaba**
Fixes #5541

## Changes

From `var config = system.Settings.Config.GetConfig("akka.channel-scheduler");` to `var config = system.Settings.Config.GetConfig("akka.actor.channel-scheduler");`

* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [x] Design discussion issue #5541

### CODE REVIEW

File: src/core/Akka.Tests/Dispatch/ChannelExecutorSpec.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public class ChannelExecutorSpec : AkkaSpec
+    {
+        [Fact]
+        public void Should_Successfully_Create_Channel_Executor_ActorSystem()
+        {
+            var config = @"
+                          akka.actor.default-dispatcher = { 
+                              executor = channel-executor
+                              channel-executor.priority = normal
```
Comment (eaba) [inline src/core/Akka.Tests/Dispatch/ChannelExecutorSpec.cs:17]: Channel executor not injected when `channel-executor.priority` is not explicitly set by user @Zetanova @Aaronontheweb 
https://github.com/akkadotnet/akka.net/blob/405dfc585e633e3bf9222ed28348c325e8d1204e/src/core/Akka/Dispatch/AbstractDispatcher.cs#L123

Comment (Zetanova) [reply src/core/Akka.Tests/Dispatch/ChannelExecutorSpec.cs:17]: I made already some improvements to config loading in PR https://github.com/akkadotnet/akka.net/pull/5515

The channel-scheduler is a standalone component and `akka.channel-scheduler` is correct,
but I copied the default config block into the wrong place in the default config:
https://github.com/akkadotnet/akka.net/blob/405dfc585e633e3bf9222ed28348c325e8d1204e/src/core/Akka/Configuration/Pigeon.conf#L272-L279

Please change the default config in `Pigeon.conf` and not `akka.channel-scheduler`

**Comment by Arkatufus** [comment]
Also need to document these enum values in the code above: https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka/Dispatch/ChannelSchedulerExtension.cs#L456-L467

File: src/core/Akka.Tests/Dispatch/ChannelExecutorSpec.cs (hunk 1 above)
Comment (Arkatufus) [reply src/core/Akka.Tests/Dispatch/ChannelExecutorSpec.cs]: This fact needs to be documented properly, it does not reflect what we have right now in the documentation here:
https://getakka.net/articles/actors/dispatchers.html#channelexecutor

**Comment by Zetanova** [comment]
I put only all names from windows process priority into the enum
They can be used but only have the same effect.

See:
https://github.com/akkadotnet/akka.net/pull/5543#discussion_r794437644

**Comment by Arkatufus** [comment]
I can't merge this, it still doesn't address the original issue

**Comment by eaba** [comment]
> I can't merge this, it still doesn't address the original issue

Yes @Arkatufus it does not. This was like showing what the issue was and luckily @Zetanova already created a PR that addresses that!

**Comment by Arkatufus** [comment]
No, his PR does not address this issue.

**Comment by eaba** [comment]
```
at Akka.Dispatch.ChannelExecutorConfigurator..ctor(Config config, IDispatcherPrerequisites prerequisites) in D:\a\1\s\src\core\Akka\Dispatch\AbstractDispatcher.cs:line 124

   at Akka.Dispatch.MessageDispatcherConfigurator.ConfigureExecutor() in D:\a\1\s\src\core\Akka\Dispatch\AbstractDispatcher.cs:line 327

   at Akka.Dispatch.DispatcherConfigurator..ctor(Config config, IDispatcherPrerequisites prerequisites) in D:\a\1\s\src\core\Akka\Dispatch\Dispatchers.cs:line 590

   at Akka.Dispatch.Dispatchers.ConfiguratorFrom(Config cfg) in D:\a\1\s\src\core\Akka\Dispatch\Dispatchers.cs:line 535

   at Akka.Dispatch.Dispatchers.LookupConfigurator(String id) in D:\a\1\s\src\core\Akka\Dispatch\Dispatchers.cs:line 439

   at Akka.Dispatch.Dispatchers..ctor(ActorSystem system, IDispatcherPrerequisites prerequisites, ILoggingAdapter logger) in D:\a\1\s\src\core\Akka\Dispatch\Dispatchers.cs:line 342

   at Akka.Actor.Internal.ActorSystemImpl.ConfigureDispatchers() in D:\a\1\s\src\core\Akka\Actor\Internal\ActorSystemImpl.cs:line 470

   at Akka.Actor.Internal.ActorSystemImpl..ctor(String name, Config config, ActorSystemSetup setup, Nullable`1 guardianProps) in D:\a\1\s\src\core\Akka\Actor\Internal\ActorSystemImpl.cs:line 102

   at Akka.Actor.ActorSystem.CreateAndStartSystem(String name, Config withFallback, ActorSystemSetup setup) in D:\a\1\s\src\core\Akka\Actor\ActorSystem.cs:line 281

   at Akka.Tests.Dispatch.ChannelExecutorSpec.Should_Successfully_Create_Channel_Executor_ActorSystem() in D:\a\1\s\src\core\Akka.Tests\Dispatch\ChannelExecutorSpec.cs:line 19
```

**Comment by Aaronontheweb** [comment]
The real issue here is that the old channel-executor HOCON still needs to work, but right now it doesn't. We need to add a backwards compat spec and make sure the old HOCON can cause the system to load the new code.

**Comment by Zetanova** [comment]
I 

> The real issue here is that the old channel-executor HOCON still needs to work, but right now it doesn't. We need to add a backwards compat spec and make sure the old HOCON can cause the system to load the new code.

@Aaronontheweb The config section of `channel-scheduler` was placed in the wrong section `akka.actor` the `Pigeon.conf`
it was always intended to be in `akka` section directly and in every post with a config example it is under `akka.channel-scheduler`

**Comment by Arkatufus** [comment]
Superseeded by #5568

**Pull request closed without merging**